src/site/_site/Documentation/styles.scss

```diff
@@ -15,3 +15,9 @@
 .Documentation__section + .Documentation__section {
   margin-top: 1rem;
 }
+
+.Documentation__section h2 {
+  margin-top: 2rem;
```

Comment:
These styles are still being added unless otherwise overwritten. So in the case of the title component an h2 would get those margins b/c we don't overwrite them.
I would suggest either writing something like `.documentation_section h2:not([class])` since all PF titles would require a class. The alternative could be leaving out styles entirely for the selector and require a class on them. It seems all the other heading selectors are using classes like `.component-heading` or `.preview-title`.